chrome/script.browserify.js

```diff
@@ -166,15 +166,28 @@ function keyHandler(e) {
 }
 
 function switchFocus(firstSelector, nextNodeAttr) {
-  var inputId = "search-field";
-  var newActive =
-    document.activeElement.id === inputId
+  var searchField = document.getElementById('search-field');
+  var newActive = document.activeElement !== searchField
       ? document.querySelector(firstSelector)
       : document.activeElement[nextNodeAttr];
 
   if (newActive) {
     newActive.focus();
   } else {
-    document.getElementById(inputId).focus();
+    searchField.focus();
   }
 }
+
+// The oncreate(vnode) hook is called after a DOM element is created and attached to the document.
+// see https://mithril.js.org/lifecycle-methods.html#oncreate for mor informations
+function oncreate() {
+  var searchField = document.getElementById('search-field');
+
+  // FireFox probably prevents `focus()` calls for some time
+  // after extension is rendered.
+  window.setTimeout(function() {
+    if (searchField) {
+      searchField.focus();
+    }
+  }, 100);
```

Comment:
there is that ugly hack with timeout:(

Reply:
Try to wrap it in `requestAnimationFrame` instead of the `setTimeout`, I'm curious if that's what FF wants.

Reply:
I'd also move the call to `getElementById` inside the `requestAnimationFrame`, just to make sure we aren't searching too early :)

Reply:
I've just found some time to try it and it doesn't work which makes sense to me. if `window.setTimeout(fc, 0)` doesn't work I wouldn't expect `requestAnimationFrame` to work either. With `0` timeout browser evaluate function right after call-stack is empty. With the request, animation frame code is evaluated even before redraw.

I also tried this sort of "recursive" approach:

```js
function oncreate() {
  var searchField = document.getElementById('search-field');
  searchField.focus();

  if (document.activeElement !== searchField) {
    window.requestAnimationFrame(oncreate);
  }
}
```
which doesn't work either.

One thing I'm still not sure about is what makes timeout `100` works. Is it guaranteed to always work (since there is some timeout before `focus()` calls are accepted) or is some race in async code? I don't really know and frankly, I don't know how to test it either. So I did it just by trying. and it seems that **with 40ms timeout it sometimes works and sometimes doesn't**. Which seems to be answer - timeout might be simply buggy.

Next thing I did was to remove whole `setTimeout` hack together with `autofocus: "on"` and test that in chrome. And it worked correctly.

To me, it seems that same thing that prevents us to simply use `autofocus` in FF is the thing that blocks `focus()` call. Maybe we can try different DOM events but I'm skeptical.

Another solution that came to my mind was to rewrite that function so it will perform check every 10ms and if it's not successful it will set itself to run in next 10ms until it is. Anyway, I did quick and it seems that `document.activeElement` returns input even though it's not really selected so there is no (easy) way to perform that test.

Reply:
Hey, thanks for doing the experiments! Then let's keep it simple and not over-engineer stuff, it's a bit hacky - yes, but it works and it's quite simple to understand, especially given that you left an explanatory comment. I'll merge now. Thanks 👍